Documentation/provisioning.md

```diff
@@ -0,0 +1,229 @@
+  3) Sends a message to the provisioning host over serial to see if it responds correctly.
+  4) Generates an Endorsement Key Certificate in the TPM using CreatePrimary and EvictControl TPM Commands.
+  5) Retrieves the Endorsement Key Certificate from the TPM using a ReadPublic command
+  6) Sends the Endorsement Key Certificate to the provisioning host.
+  7) Requests the cross-signed version of the Endorsement Key Certificate from the provisioning host and saves it into a UEFI variable.
+  8) Requests per-device SMBIOS customizations from the provisioning host and saves each one in its own UEFI variable.
+  9) Sets "DeviceProvisioned" so that all subsequent boots will return from this driver immediately.
+
+## Device side code and board specific customizations
+
+### U-Boot SPL
+
+U-Boot SPL is responsible for fusing the SRKH High Assurance Boot Key and the MAC address into SoC fuses.
+
+1) Copy the code within `#ifdef CONFIG_SPL_BOARD_PROVISION` from board\solidrun\mx6cuboxi\mx6cuboxi.c into your own board specific code. This code includes struct srkfdt and *spl_board_provision( )*.
+
+2) Set CONFIG_SPL_BOARD_PROVISION=y in your board defconfig
+
+3) Build the firmware, deploy it to the device, start provision.py from the "imx-iotcore" section below, and attempt to boot. This firmware will not blow any of the fuses, it will just print out what it would blow. This can be used to verify that the SRKH and MAC addresses are correct before permanently committing them to the platform.
+
+4) Once you've verified that the values are correct you can set CONFIG_SPL_BOARD_PROVISION_FUSES=y in your board defconfig so that *spl_board_provision( )* will blow the SRKH and MAC fuses.
+
+### OP-TEE
+
+The changes in OP-TEE are responsible for making sure the SoC will only ever provision a single eMMC RPMB with its secret key.
+
+1) Add CFG_FSL_SEC=y and CFG_RPMB_KEY_WRITE_LOCK=y to your OP-TEE flags in your board makefile. [Example Makefile](build\firmware\HummingBoardEdge_iMX6Q_2GB\Makefile)
+
+2) If any portion of the GP1 or GP2 fuse words are already reserved for another use on your board then customize which fuse bit to use for the RPMB write lock. Open `core/drivers/fsl_sec/hw_key_blob.c` and modify both `RPMB_KEY_WRITE_LOCK_FUSE_BITS` and `RPMB_KEY_WRITE_LOCK_FUSE_WORD`
+
+3) When the fTPM tries to access the RPMB for the first time, OP-TEE will run through this flow and blow the fuse if the RPMB key write was successful. If the RPMB key is already provisioned, then this flow is skipped entirely.
+
+### UEFI
+
+The changes in UEFI are responsible for the following:
+* Loading SMBIOS values from UEFI variables.
+* Create an fTPM Endorsement Key Certificate using the default RSA template.
+* Loading the Endorsement Key Certificate from the fTPM and saving it to a host computer.
+* Receiving  a cross-signed EK Certificate to save into UEFI variables on the device for easy access.
```

Comment:
Do we want to store this as a variable, or back in the TPM? I had a quick look and I wasn't able to find an answer online, so we can probably run with this for now and wait for an expert opinion.

Reply:
Either way presents a compelling case. Leaving it as is for now and will reconsider later.